AWBWApp.Game/Editor/CaptureCalcHelper.cs

```diff
@@ -0,0 +1,399 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using AWBWApp.Game.API.Replay;
+using AWBWApp.Game.Game.Building;
+using AWBWApp.Game.Game.Country;
+using AWBWApp.Game.Game.Logic;
+using AWBWApp.Game.Game.Tile;
+using AWBWApp.Game.Game.Units;
+using AWBWApp.Game.Helpers;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Bindables;
+
+namespace AWBWApp.Game.Editor
+{
+    public class CapStop
+    {
+        public int extraTurns = 0; // Defines how many turns we have already looked ahead to try to find another cap stop
+        public Vector2I coord;
+        public CapStop(Vector2I coord)
+        {
+            this.coord = coord;
+        }
+        public override String ToString()
+        {
+            return coord + "+" + extraTurns;
+        }
+    }
+
+    public class CapPhaseAnalysis
+    {
+        public Dictionary<Vector2I, List<List<CapStop>>> capChains = new Dictionary<Vector2I, List<List<CapStop>>>();
+        public List<Vector2I> contestedProps = new List<Vector2I>(); // as was probably considered by the map designer; doesn't necessarily take movement/production differences into account
+
+        public override String ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.Append("Contested properties:");
+            foreach( Vector2I contested in contestedProps )
+                sb.Append(String.Format(" %s ", contested));
+            sb.Append("\n");
+            foreach( Vector2I factoryXYC in capChains.Keys )
+            {
+                sb.Append(String.Format("Cap chains for %s:\n", factoryXYC));
+                foreach( List<CapStop> chain in capChains[factoryXYC] )
```

Comment:
```suggestion
            foreach (var chainPair in capChains)
            {
                sb.AppendLine($"Cap chains for {chainPair.Key}");
                foreach (var chain in chainPair.Value)
```

Reply:
This actively makes the code less meaningful.
`chainPair` is just type information - `factoryXYC` tells you that this specifically refers to a *factory*.
The "XYC" could/should probably change since you're using vectors instead of a custom class called `XYCoord`, to be fair.